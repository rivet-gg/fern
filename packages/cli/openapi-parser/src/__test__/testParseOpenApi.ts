import { AbsoluteFilePath, join, RelativeFilePath } from "@fern-api/fs-utils";
import { CONSOLE_LOGGER } from "@fern-api/logger";
import { serialization } from "@fern-api/openapi-ir-sdk";
import { createMockTaskContext } from "@fern-api/task-context";
import { parse, SpecImportSettings } from "../parse";

const FIXTURES_PATH = join(AbsoluteFilePath.of(__dirname), RelativeFilePath.of("fixtures"));

// eslint-disable-next-line jest/no-export
export function testParseOpenAPI(
    fixtureName: string,
    openApiFilename: string,
    asyncApiFilename?: string,
    settings?: SpecImportSettings
): void {
    // eslint-disable-next-line jest/valid-title
    describe(fixtureName, () => {
        it("parse open api", async () => {
            const absolutePathToOpenAPI = join(
                FIXTURES_PATH,
                RelativeFilePath.of(fixtureName),
                RelativeFilePath.of(openApiFilename)
            );
            const absolutePathToAsyncAPI =
                asyncApiFilename != null
                    ? join(FIXTURES_PATH, RelativeFilePath.of(fixtureName), RelativeFilePath.of(asyncApiFilename))
                    : undefined;
            const specs = [];
            if (absolutePathToOpenAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToOpenAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }
            if (absolutePathToAsyncAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToAsyncAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }

            const openApiIr = await parse({
                specs,
                taskContext: createMockTaskContext({ logger: CONSOLE_LOGGER })
            });
            const openApiIrJson = await serialization.OpenApiIntermediateRepresentation.jsonOrThrow(openApiIr, {
                skipValidation: true
            });
            expect(openApiIrJson).toMatchSnapshot();
        });
    });
}
