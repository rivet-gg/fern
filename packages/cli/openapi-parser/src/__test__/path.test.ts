import { testParseOpenAPI } from "./testParseOpenApi";

describe("path parsing", () => {
    testParseOpenAPI("path", "openapi.yml");
});
