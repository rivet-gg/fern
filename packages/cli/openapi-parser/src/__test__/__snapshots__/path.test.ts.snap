// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`path parsing path parse open api 1`] = `
{
  "apiVersion": null,
  "basePath": null,
  "channel": [],
  "description": null,
  "endpoints": [
    {
      "audiences": [],
      "authed": false,
      "availability": null,
      "description": null,
      "errors": {},
      "examples": [
        {
          "codeSamples": [],
          "description": null,
          "headers": [],
          "name": null,
          "pathParameters": [],
          "queryParameters": [],
          "request": null,
          "response": null,
          "type": "full",
        },
      ],
      "generatedRequestName": "GetExampleWithoutTrailingSlashRequest",
      "headers": [],
      "idempotent": null,
      "internal": null,
      "method": "GET",
      "operationId": null,
      "pagination": null,
      "path": "/example-without-trailing-slash",
      "pathParameters": [],
      "queryParameters": [],
      "request": null,
      "requestNameOverride": null,
      "response": null,
      "sdkName": null,
      "server": [],
      "summary": "Get Example",
      "tags": [],
    },
    {
      "audiences": [],
      "authed": false,
      "availability": null,
      "description": null,
      "errors": {},
      "examples": [
        {
          "codeSamples": [],
          "description": null,
          "headers": [],
          "name": null,
          "pathParameters": [],
          "queryParameters": [],
          "request": null,
          "response": null,
          "type": "full",
        },
      ],
      "generatedRequestName": "GetExampleWithTrailingSlashRequest",
      "headers": [],
      "idempotent": null,
      "internal": null,
      "method": "GET",
      "operationId": null,
      "pagination": null,
      "path": "/example-with-trailing-slash/",
      "pathParameters": [],
      "queryParameters": [],
      "request": null,
      "requestNameOverride": null,
      "response": null,
      "sdkName": null,
      "server": [],
      "summary": "Get Example with trailing",
      "tags": [],
    },
  ],
  "globalHeaders": [],
  "groups": {},
  "hasEndpointsMarkedInternal": false,
  "idempotencyHeaders": [],
  "nonRequestReferencedSchemas": [],
  "schemas": {},
  "securitySchemes": {},
  "servers": [],
  "tags": {
    "orderedTagIds": null,
    "tagsById": {},
  },
  "title": "Sample Path Parsing API",
  "variables": {},
  "webhooks": [],
}
`;
